### ISSUE TITLE
talisker 0.9.1 breaks with older setuptools

### ISSUE THREAD

**Issue by thomir** [canonical-ols/talisker#65]
Talisker 0.9.1's celery entry point fails - I think when run under python3:

```
[2;34m2017-03-24 02:31:33.190Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:33.192Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:33.559Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:33.561Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:33.922Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:33.924Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:34.283Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:34.284Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:34.624Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:34.625Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:34.962Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:34.963Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:35.297Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:35.298Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:35.657Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:35.659Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:35.999Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:36.000Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:36.331Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:36.332Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:36.680Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:36.681Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:37.038Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:37.040Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:37.385Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:37.387Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:37.739Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:37.740Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:38.061Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:38.062Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:38.409Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:38.410Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:38.759Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:38.760Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:39.117Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:39.118Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:39.466Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:39.467Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
[2;34m2017-03-24 02:31:39.808Z [0m[0;32mINFO[0m [0;33mtalisker.sentry.TaliskerSentryClient [0m"[1;37mRaven is not configured (logging is disabled). Please see the documentation for more information.[0m"
[2;34m2017-03-24 02:31:39.810Z [0m[0;32mINFO[0m [0;33mtalisker.statsd [0m"[1;37mconfiguring statsd via environment[0m" [3;36mSTATSD_DSN=udp://localhost:8125/[0m
Traceback (most recent call last):
  File "./env/bin/talisker.celery", line 11, in <module>
    sys.exit(main())
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/celery.py", line 177, in main
    ensure_extra_versions_supported('celery')
  File "/srv/snap-delta-service/production/snap-delta-service/env/lib/python3.4/site-packages/talisker/util.py", line 55, in ensure_extra_versions_supported
    if pkg.version not in requirement.specifier:
AttributeError: 'Requirement' object has no attribute 'specifier'
```

We've reverted to 0.9.0 for the time being.

**Comment by bloodearnest** [comment]
It's not python3, it's the version of pkg_resources. In a venv with modern setuptools it's fine, but looks like the version of pkg_resources in whatever version of setuptools your using doesn't support specifiers. I will fix it, as i still want to support the version of pip/setuptools in trusty.

**Comment by thomir** [comment]
Thanks @bloodearnest - I wasn't sure what the problem was. As you've guessed, this is for a deployment based on trusty.

**Comment by bloodearnest** [comment]
Fixed in #71
